Updated the PWA Badging How-To page based on a new WebKit blog

kevinctofel opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

- Updated PWA Badging to reflect new WebKit support requires notification permissions.
- Explains how to request notification permissions and how to check for prior notification permission grants before attempting to show badging notifications.

<!-- ✍️ Summarize your changes in one or two sentences -->

### Motivation

This update provides information, documentation and links developers need before implementing PWA Badges on Safari and any other mobile WebKit-based browsers.

<!-- ❓ Why are you making these changes and how do they help readers? -->

### Additional details

Source information from Apple: https://webkit.org/blog/14112/badging-for-home-screen-web-apps/
Note: This is my first ever PR / contribution so I welcome constructive criticism. 

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests

Fixes #26430 

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

kevinctofel committed:
Updated docs with WebKit information

estelle commented on files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md:
```diff
@@ -29,7 +29,7 @@ Safari and Firefox on desktop do not support the Badging API and do not support
 ### Mobile support
 
-On mobile operating systems, badges are supported on Chromium-based browsers running on Android.
+On mobile operating systems, badges are supported on Chromium-based browsers running on Android and on WebKit-based browsers, running iOS or iPadOS 16.4 and up.
```
```suggestion
Badges are supported on mobile operating systems, including Chromium-based browsers running on Android and iOS and iPadOS, starting with iPadOS16.4.
```

kevinctofel replied:
Good call and that's for spotting! You're correct that PWAs on iOS are currently only supported with Safari. Verfied at: https://web.dev/learn/pwa/installation/#ios-and-ipados-installation and I also installed / tested on Brave and Chrome for iOS. No PWA installs for either of those as expected. Committing this change.

estelle commented on files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md:
```diff
@@ -49,6 +49,18 @@ if (navigator.setAppBadge) {
 
+### Request notification permissions for iOS and/or iPadOS
+
+While notification badges are supported on WebKit, badges will not appear until your application is granted notification permissions. To request notification permissions, call the [Notification.requestPermission()](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API#getting_permission) method, such as in the following basic example:
```
```suggestion
While notification badges are supported on iOS and iPadOS, badges will not appear until the application is granted notification permissions. To request notification permissions, call the [Notification.requestPermission()](/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API#getting_permission) method:
```

estelle commented on files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md:
````diff
@@ -49,6 +49,18 @@ if (navigator.setAppBadge) {
+});
+
+```
+Optionally, you can check if a user has previously granted notification permissions using the [Permissions API](https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API).
````
```suggestion
Optionally, you can check if a user has previously granted notification permissions using the [Permissions API](/en-US/docs/Web/API/Permissions_API).
```

estelle reviewed (commented):
Thank you for the edits. I made a few suggestions.

For in-MDN links, we drop the `https://developer.mozilla.org/`, and start the href with `/en-US/docs`

kevinctofel committed:
Update files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md

Updated to reflect PWA installs only supported in Safari on iOS and iPadOS.

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

kevinctofel committed:
Update files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md

Now reflects WebKit in lieu of iOS/iPadOS and shortened MDN link per style suggestion.

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

kevinctofel committed:
Update files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md

Shortened MDN link per style suggestion.

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

estelle reviewed (approved):
Thank you. Also, congratulations on your first merged MDN content PR. Welcome to he team!
